implementation/endpoints/src/tcp_client_endpoint_impl.cpp

```diff
@@ -123,6 +115,7 @@ void tcp_client_endpoint_impl::restart(bool _force) {
                         << std::setw(4) << its_session << "]"
                         << " size: " << std::dec << q.first->size();
             }
+            self->sending_blocked_ = false;
```

Comment:
I described in #668 that there's a defect where `is_sending_` needs to get cleared. Clearing `sending_blocked_` is not sufficient to fix that bug.

I suspect that _both_ flags might need to get cleared during restart, but what is the rationale for setting `sending_blocked_` false here? What was the bug?

Reply:
Hi @joeyoravec, as far as I know this was a "bug" where we were ignoring StopOffer messages (Fix to not ignore stop offers when sd acceptance is not required), however this pr was causing a lot of other problems and was reverted in 3.5.1. You can my check other pr in draft.
We will have to look at your issue #668.